### DISCUSSION TITLE
Write data to Berry list object, but no as a string

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#21838] [Q&A] [answered]
Hello!

I want plot sensor values in a "timeofday" Google Chart. For this i save the values (hour, minute, second) and the temperature in an list. The webserver module then sends the html code and the list to the client. The Javascript Function expects the data in the following format (code from the html file that works!):

```c
function drawChart() {
     var data = google.visualization.arrayToDataTable([
       [{type: 'timeofday', label: 'Zeit'}, 'Temperatur'],
       [[8, 0, 0],  24],
       [[9, 0, 0],  25],
       [[10, 0, 0],  26],
       [[11, 0, 0],  27]
     ]);
```

For this in Berry i generate the following list:

```c
var data = []

data.push([{type: 'timeofday', label: 'Zeit'}, 'Temperatur'])
data.push([[8, 0, 0],  24])
data.push([[9, 0, 0],  25])
data.push([[10, 0, 0], 26])
data.push([[11, 0, 0], 27])
```

and send the list and html-code memory efficient with the Berry webserver module to the client (only the interesting code part):
    
```c
webserver.content_send(

f"<html>"
"<head>"
    
"<script type='text/javascript' src='https://www.gstatic.com/charts/loader.js'></script><script type='text/javascript'>google.charts.load('current', {{'packages':['corechart']}});google.charts.setOnLoadCallback(drawChart);"
"function drawChart() {{var data = google.visualization.arrayToDataTable({data});"
"var options = {{title: 'Company Performance',hAxis: {{title: 'Year',  titleTextStyle: {{color: '#333'}}}},vAxis: {{minValue: 0}}}};"
"var chart = new google.visualization.AreaChart(document.getElementById('chart_div'));chart.draw(data, options);}}"
"</script>"

"</head>"
"<body>"
"<div id='chart_div' style='width: 100%; height: 500px;'></div>"
"</body>"
"</html>"
)
```

So, now my problem is, that in Berry this line doesn't work:

```c
data.push([{type: 'timeofday', label: 'Zeit'}, 'Temperatur'])
```
The following as string works, but then it doesn't have the right format for the google.visualization.arrayToDataTable function:

`data.push("[{label: \'Date\', type: \'timeofday\'}, \'Sales\', \'Expenses\']")`

output:

```c
print(data)
['[{label: \'Date\', type: \'timeofday\'}, \'Sales\', \'Expenses\']']
```

My question now: is there a way to write the "configuration" in the first row of the list, but no as a string?

Thank you!

**Comment by sfromis** [comment, marked as answer]
Things may get simpler if you take advantage of Google Charts being flexible about syntax. Those keywords (like `label`)  can be strings, instead of unquoted words. Hence, this syntax is workable, and then you can have the resulting string directly in your `google.visualization.arrayToDataTable(xxxx)`

```
    var data = []
    data.push([{'type': 'timeofday', 'label': 'Zeit'}, 'Temperatur'])
    data.push([[8, 0, 0],  24])
    data.push([[9, 0, 0],  25])
    data.push([[10, 0, 0], 26])
    data.push([[11, 0, 0], 27])
    var s = str(data)
```

**Reply by ghost** [reply]
Works great! Thank you!
